d2graph/d2graph.go

```diff
@@ -951,32 +980,35 @@ func (g *Graph) SetDimensions(mtexts []*d2target.MText, ruler *textmeasure.Ruler
 		dims.Width += innerLabelPadding
 		dims.Height += innerLabelPadding
 		obj.LabelDimensions = *dims
-		obj.Width = float64(dims.Width)
-		obj.Height = float64(dims.Height)
+		// the desired dimensions must be at least as large as the text
+		obj.Width = float64(go2.Max(dims.Width, desiredWidth))
+		obj.Height = float64(go2.Max(dims.Height, desiredHeight))
 
-		switch strings.ToLower(obj.Attributes.Shape.Value) {
+		switch shapeType {
 		default:
-			obj.Width += 100
-			obj.Height += 100
+			if desiredWidth == 0 {
+				obj.Width += 100
+			}
+			if desiredHeight == 0 {
+				obj.Height += 100
+			}
 
 		case d2target.ShapeImage:
-			if obj.Attributes.Width != nil {
-				w, _ := strconv.Atoi(obj.Attributes.Width.Value)
-				obj.Width = float64(w)
-			} else {
+			if desiredWidth == 0 {
 				obj.Width = 128
 			}
-			if obj.Attributes.Height != nil {
-				h, _ := strconv.Atoi(obj.Attributes.Height.Value)
-				obj.Height = float64(h)
-			} else {
+			if desiredHeight == 0 {
 				obj.Height = 128
 			}
 
 		case d2target.ShapeSquare, d2target.ShapeCircle:
 			sideLength := go2.Max(obj.Width, obj.Height)
-			obj.Width = sideLength + 100
-			obj.Height = sideLength + 100
+			padding := 0.
+			if desiredWidth == 0 && desiredHeight == 0 {
```

Comment:
i guess squares and circles should actually have special handling. where u can just set one and it's assumed the other matches

Reply:
yup just confirming this is the behavior